tests/test_integration.py

```diff
@@ -82,150 +109,166 @@ def _create_vm():
         vm_obj.delete()
         vm_obj.commit()
 
-    vm_obj = create({
-        'hostname': VM1_host,
-        'project_network': VM1_net,
-        'servertype': 'vm',
-        'project': 'test',
-        'team': 'test',
-        'xen_host': HV1,
-    })
-    vm_obj.commit()
 
-    # Query for it from igvm.vm and reset to defaults
-    vm = VM(VM1_host)
-    vm.server_obj['intern_ip'] = IP1
-    vm.server_obj['state'] = 'online'
-    vm.server_obj['disk_size_gib'] = 3
-    vm.server_obj['memory'] = 2048
-    vm.server_obj['num_cpu'] = 2
-    vm.server_obj['os'] = 'jessie'
-    vm.server_obj['environment'] = 'testing'
-    vm.server_obj['repositories'] = [
-        'int:basejessie:stable',
-        'int:innogames:stable jessie',
-    ]
-    if 'puppet_environment' in vm.server_obj:
-        vm.server_obj['puppet_environment'] = None
-    vm.server_obj.commit()
-    # Might have changed!
-    vm.hypervisor = Hypervisor(vm.server_obj['xen_host'])
-    return vm
-
-
-def _clean_vm(hvs, vm):
-    for hv in hvs:
-        hv.run(
-            'virsh destroy {vm}; '
-            'virsh undefine {vm}'
-            .format(vm=vm.fqdn),
-            warn_only=True,
-        )
-        hv.run(
-            'umount /dev/xen-data/{vm}; '
-            'lvremove -f /dev/xen-data/{vm}'
-            .format(vm=vm.fqdn),
-            warn_only=True,
-        )
+def cmd(cmd, *args, **kwargs):
+    escaped_args = [quote(str(arg)) for arg in args]
 
+    escaped_kwargs = {}
+    for key, value in kwargs.iteritems():
+        escaped_kwargs[key] = quote(str(value))
 
-buildvm = partial(real_buildvm, balance_config='test')
-migratevm = partial(real_migratevm, balance_config='test')
+    return cmd.format(*escaped_args, **escaped_kwargs)
 
 
 class IGVMTest(unittest.TestCase):
-    def _check_vm(self, hv, vm):
-        fqdn = vm.run('hostname -f').strip()
-        self.assertEqual(fqdn, vm.fqdn)
-
-        self.assertEqual(vm.server_obj.is_dirty(), False)
-
-        self.assertEqual(hv.vm_defined(vm), True)
-        self.assertEqual(hv.vm_running(vm), True)
-
-    def _check_absent(self, hv, vm):
-        self.assertEqual(hv.vm_defined(vm), False)
-        hv.run('test ! -b /dev/xen-data/{}'.format(vm.fqdn))
-
-
-class BalanceBuildTest(IGVMTest):
-    """ Same as BuildTest but with tearDown and setUp cleaning *both* HVs """
-
     @classmethod
     def setUpClass(cls):
-        cls.hv1 = Hypervisor(HV1)
-        cls.hv2 = Hypervisor(HV2)
-        cls.vm = _create_vm()
+        cls.hvs = HYPERVISORS
```

Comment:
I rather hoped to never have any global variables at all, but since I found no better way to pass variable between `setUpModule()` and the rest of code, we go fully global now.

Reply:
Sure, not having globals is of course the best.